analytics.go

```diff
@@ -239,10 +243,28 @@ func (c *Client) Close() error {
 	return nil
 }
 
+func (c *Client) sendAsync(msgs []interface{}) {
+	c.inflight++
+	go func() {
+		c.sendDone <- c.send(msgs)
+	}()
+}
+
+func (c *Client) awaitInflight() {
+	for c.inflight > 0 {
+		c.verbose("waiting for %d requests to finish..", c.inflight)
+		err := <-c.sendDone
+		if err != nil {
+			c.logf(err.Error())
+		}
+		c.inflight--
+	}
+}
```

Comment:
Aye, I see your point. Let me see if I can get the waitgroup stuff in before I have to run out the door.